yield_plugin/src/yield_plugin.cpp

```diff
@@ -263,55 +263,66 @@ namespace yield_plugin
+      }
 
-    // seperating cooperative yield with regular object detection for better performance.
-    if (config_.enable_cooperative_behavior && clc_urgency_ > config_.acceptable_urgency)
-    {
-      RCLCPP_DEBUG(nh_->get_logger(),"Only consider high urgency clc");
-      if (timesteps_since_last_req_ < config_.acceptable_passed_timesteps)
+      // seperating cooperative yield with regular object detection for better performance.
+      if (config_.enable_cooperative_behavior && clc_urgency_ > config_.acceptable_urgency)
       {
-        RCLCPP_DEBUG(nh_->get_logger(),"Yield for CLC. We haven't received an updated negotiation this timestep");
-        yield_trajectory = update_traj_for_cooperative_behavior(original_trajectory, initial_velocity);
-        timesteps_since_last_req_++;
+        RCLCPP_DEBUG(nh_->get_logger(),"Only consider high urgency clc");
+        if (timesteps_since_last_req_ < config_.acceptable_passed_timesteps)
+        {
+          RCLCPP_DEBUG(nh_->get_logger(),"Yield for CLC. We haven't received an updated negotiation this timestep");
+          yield_trajectory = update_traj_for_cooperative_behavior(original_trajectory, initial_velocity);
+          timesteps_since_last_req_++;
+        }
+        else
+        {
+          RCLCPP_DEBUG(nh_->get_logger(),"unreliable CLC communication, switching to object avoidance");
+          yield_trajectory = update_traj_for_object(original_trajectory, external_objects_, initial_velocity); // Compute the trajectory
+        }
       }
       else
       {
-        RCLCPP_DEBUG(nh_->get_logger(),"unreliable CLC communication, switching to object avoidance");
+        RCLCPP_DEBUG(nh_->get_logger(),"Yield for object avoidance");
         yield_trajectory = update_traj_for_object(original_trajectory, external_objects_, initial_velocity); // Compute the trajectory
       }
-    }
-    else
-    {
-      RCLCPP_DEBUG(nh_->get_logger(),"Yield for object avoidance");
-      yield_trajectory = update_traj_for_object(original_trajectory, external_objects_, initial_velocity); // Compute the trajectory
-    }
 
-    // return original trajectory if no difference in trajectory points a.k.a no collision
-    if (fabs(get_trajectory_end_time(original_trajectory) - get_trajectory_end_time(yield_trajectory)) < EPSILON)
-    {
-      resp->trajectory_plan = original_trajectory;
+      // return original trajectory if no difference in trajectory points a.k.a no collision
+      if (fabs(get_trajectory_end_time(original_trajectory) - get_trajectory_end_time(yield_trajectory)) < EPSILON)
+      {
+        resp->trajectory_plan = original_trajectory;
+      }
+      else
+      {
+        yield_trajectory.header.frame_id = "map";
+        yield_trajectory.header.stamp = nh_->now();
+        yield_trajectory.trajectory_id = original_trajectory.trajectory_id;
+        resp->trajectory_plan = yield_trajectory;
+      }
     }
-    else
+    catch(const std::runtime_error& e)
```

Comment:
Is there a more specific error we can catch here. Sonar scanner discourages catching generic exceptions for error handling and encourages catching more specific ones.

Reply:
https://sonarcloud.io/project/issues?sinceLeakPeriod=true&issueStatuses=OPEN%2CCONFIRMED&pullRequest=2502&id=usdot-fhwa-stol_CARMAPlatform&open=AZVEDtGbGJGytsS7SHKc

Reply:
There is no specific error we can catch. Please look at my comment above. the error thrown is itself runtime_error

Reply:
Okay I figured out a way to catch the specific error based on the error message not the error type.

Reply:
If that is the case I actually prefer you change it back to what you had originally. The idea of throwing specific exceptions and catching is to allow error handling to respond differently based on the exception thrown. It sounds like we should throw a more specific exception and then catch that specific one here but for the sake of this PR I would prefer if you revert the change looking for a specific error message. That is not very maintainable.